File: backend/app/services/factory_service.py
from sqlalchemy.orm import Session

from app.models.factory import Factory
from app.repositories.factory_repository import (
    create_factory,
    delete_factory,
    get_factories,
    get_factory,
    update_factory,
)
from app.schemas.factory import FactoryCreate


def create_factory_service(
    db: Session,
    data: FactoryCreate,
) -> Factory:
    factory = Factory(**data.model_dump())
    return create_factory(db, factory)


def get_factories_service(
    db: Session,
) -> list[Factory]:
    return get_factories(db)


def get_factory_service(
    db: Session,
    factory_id: int,
) -> Factory | None:
    return get_factory(db, factory_id)


def update_factory_service(
    db: Session,
    factory_id: int,
    data: FactoryCreate,
) -> Factory | None:
    factory = get_factory(db, factory_id)

    if not factory:
        return None

    for field, value in data.model_dump().items():
        setattr(factory, field, value)

    return update_factory(db, factory)


def delete_factory_service(
    db: Session,
    factory_id: int,
) -> bool:
    factory = get_factory(db, factory_id)

    if not factory:
        return False

    delete_factory(db, factory)
    return True

File: frontend/src/services/factoryService.js
import api from "./api";

export const getFactories = async () => {
  const response = await api.get("/factories/");
  return response.data;
};

export const getFactory = async (factoryId) => {
  const response = await api.get(`/factories/${factoryId}`);
  return response.data;
};

export const createFactory = async (data) => {
  const response = await api.post("/factories/", data);
  return response.data;
};

export const updateFactory = async (factoryId, data) => {
  const response = await api.put(`/factories/${factoryId}`, data);
  return response.data;
};

export const deleteFactory = async (factoryId) => {
  const response = await api.delete(`/factories/${factoryId}`);
  return response.data;
};